pyhive/hive.py

```diff
@@ -116,7 +124,16 @@ def sasl_factory():
                     raise AssertionError
                 sasl_client.init()
                 return sasl_client
-            self._transport = thrift_sasl.TSaslClientTransport(sasl_factory, sasl_auth, socket)
+		
+            if mode == "HTTP" and auth == "NONE":
```

Comment:
No reason to re-check auth here? Also, there's no overlap with the SASL code, so this would be a bit cleaner in the outer if, like the `if auth == 'NOSASL':` branch.

nitpick: convention in this project is single quotes for code-like strings, double quotes for human language strings.